Engine/DataFeeds/Enumerators/Factories/CorporateEventEnumeratorFactory.cs

```diff
@@ -80,34 +80,17 @@ public static class CorporateEventEnumeratorFactory
             MapFileResolver mapFileResolver)
         {
             var mapFileToUse = new MapFile(config.Symbol.Value, new List<MapFileRow>());
+            var hasUnderlying = config.Symbol.HasUnderlying;
 
-            // load up the map and factor files for equities
-            if (!config.IsCustomData && config.SecurityType == SecurityType.Equity)
+            // load up the map and factor files for equities, options, and custom data
+            if (config.UsesMapFiles)
```

Comment:
love it! 👍

Reply:
This might be a bit problematic in that Options will be allowed through the gate, even though in the past it seemed to be only for equities. Either we're introducing a bug unknowingly, or fixing one. I'd like to review this and the following conditions further to ensure that the current implementation was intended.

`Engine/DataFeeds/SubscriptionDataReader.cs ` - Lines 225-226 
`Engine/HistoricalData/SubscriptionDataReaderHistoryProvider.cs` - Line 111

Reply:
Great catch!!

Options are a whole 'nother beast. There is an `OptionRenameRegressionAlgorithm` which seems to imply that we've thought about this in the past.

This makes me somewhat concerned about potentially other locations that may suffer from a similar situation.